classes/SenseiManager.php

```diff
@@ -32,67 +34,99 @@ public static function getStudentBoost(int $graduated_count): array {
             $sensei_list[] = [
                 'sensei_id' => $result['sensei_id'],
                 'students' => json_decode($result['students']),
+                'temp_students' => json_decode($result['temp_students'], true),
                 'recruitment_message' => $result['recruitment_message'],
                 'graduated' => $result['graduated_count'],
                 'specialization' => $result['specialization'],
+                'enable_lessons' => $result['enable_lessons'],
 			    'avatar_link' => $result['avatar_link'],
 			    'user_name' => $result['user_name'],
-			    'bloodline_name' => $result['bloodline_name']
+                'ninjutsu_skill' => $result['ninjutsu_skill'],
+                'taijutsu_skill' => $result['taijutsu_skill'],
+                'genjutsu_skill' => $result['genjutsu_skill'],
+                'bloodline_skill' => $result['bloodline_skill'],
+                'speed' => $result['speed'],
+                'cast_speed' => $result['cast_speed'],
+                'exp' => $result['exp'],
             ];
         }
 		foreach ($sensei_list as &$sensei) {
-            $sensei += SenseiManager::getStudentBoost($sensei['graduated'], $system);
+            $sensei += SenseiManager::getStudentBoostBySensei($sensei['sensei_id'], $system);
 			$sensei['students'] = SenseiManager::getStudentData($sensei['students'], $system);
+            $sensei['temp_students'] = SenseiManager::checkTempStudents($sensei['sensei_id'], $system);
+            $sensei['temp_students'] = SenseiManager::getTempStudentData($sensei['temp_students'], $system);
         }
         return $sensei_list;
     }
 
     public static function getSenseiByID(int $sensei_id, System $system): array {
         $sensei = [];
         $sensei_table_result = $system->db->query(
-            "SELECT `students`, `recruitment_message`, `student_message`, `graduated_count`, `specialization` FROM `sensei` WHERE `sensei_id` = '{$sensei_id}'"
+            "SELECT * FROM `sensei` WHERE `sensei_id` = '{$sensei_id}'"
         );
         $result = $system->db->fetch($sensei_table_result);
         $sensei = array(
             'sensei_id' => $sensei_id,
             'students' => json_decode($result['students']),
             'recruitment_message' => $result['recruitment_message'],
-            'student_message' => $result['student_message'],
             'graduated' => $result['graduated_count'],
-            'specialization' => $result['specialization']
+            'student_message' => $result['student_message'],
+            'specialization' => $result['specialization'],
+            'graduated_students' => json_decode($result['graduated_students']),
+            'temp_students' => json_decode($result['temp_students'], true),
+            'time_trained' => $result['time_trained'],
+            'yen_gained' => $result['yen_gained'],
+            'is_active' => $result['is_active'],
+            'enable_lessons' => $result['enable_lessons'],
```

Comment:
We should make a `SenseiInfoDto` for this, much easier to work with outside this function

Reply:
Agreed. I'm thinking this is a to-do as part of cleaning up the sensei functionality and converting to API.